GUI/Types/Renderer/HitboxSetSceneNode.cs

```diff
@@ -0,0 +1,202 @@
+
+            for (var i = 0; i < hitboxSet.Length; i++)
+            {
+                var hitbox = hitboxSet[i];
+                sceneNodes[i] = CreateSceneNode(Scene, hitbox);
+
+                if (string.IsNullOrEmpty(hitbox.BoneName))
+                {
+                    hitboxBoneIndexes[i] = -1;
+                }
+                else
+                {
+                    hitboxBoneIndexes[i] = boneIndexes[hitbox.BoneName];
+                }
+            }
+
+            var data = new HitboxSetData
+            {
+                HitboxSet = hitboxSet,
+                HitboxBoneIndexes = hitboxBoneIndexes,
+                SceneNodes = sceneNodes
+            };
+
+            hitboxSets.Add(name, data);
+        }
+
+        private static Color32 GetHitboxGroupColor(int group)
+        {
+            if (group < 0 || group >= HitboxColors.Length)
+            {
+                return HitboxColors[0];
+            }
+            return HitboxColors[group];
+        }
+
+        private static HitboxSceneNode CreateSceneNode(Scene scene, Hitbox hitbox)
+        {
+            var color = GetHitboxGroupColor(hitbox.GroupId);
+            return hitbox.ShapeType switch
+            {
+                Hitbox.HitboxShape.Sphere => HitboxSceneNode.CreateSphereNode(scene, hitbox.MinBounds, hitbox.ShapeRadius, color),
+                Hitbox.HitboxShape.Capsule => HitboxSceneNode.CreateCapsuleNode(scene, hitbox.MinBounds, hitbox.MaxBounds, hitbox.ShapeRadius, color),
+                Hitbox.HitboxShape.Box => HitboxSceneNode.CreateBoxNode(scene, hitbox.MinBounds, hitbox.MaxBounds, color),
+                _ => throw new NotImplementedException($"Unknown hitbox shape type: {hitbox.ShapeType}")
+            };
+        }
+
+        public void SetHitboxSet(string set)
+        {
+            if (set == null)
+            {
+                currentSet = null;
+                return;
+            }
+
+            currentSet = hitboxSets[set];
+        }
+
+        public override void Update(Scene.UpdateContext context)
+        {
+            if (currentSet == null)
+            {
+                return;
+            }
+
+            LocalBoundingBox = new AABB(new Vector3(float.MinValue), new Vector3(float.MaxValue));
+
+            Frame frame = null;
+
+            if (animationController.ActiveAnimation != null)
+            {
+                if (animationController.IsPaused)
+                {
+                    frame = animationController.FrameCache.GetFrame(animationController.ActiveAnimation, animationController.Frame);
+                }
+                else
+                {
+                    frame = animationController.FrameCache.GetInterpolatedFrame(animationController.ActiveAnimation, animationController.Time);
+                }
+            }
+
+            foreach (var root in skeleton.Roots)
+            {
+                GetAnimationMatrixRecursive(root, Matrix4x4.Identity, frame);
+            }
```

Comment:
Surely these can be in AnimationController. We will almost certainly need it elsewhere too.